From 1721d69cc7d8d52b280d48db0925d5a6358fe3bd Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Tue, 20 May 2025 19:36:02 +0200
Subject: [PATCH 228/254] support: Use const char * argument in
 support_capture_subprogram_self_sgid

The function does not modify the passed-in string, so make this clear
via the prototype.

Reviewed-by: Carlos O'Donell <carlos@redhat.com>
(cherry picked from commit f0c09fe61678df6f7f18fe1ebff074e62fa5ca7a)
---
 support/capture_subprocess.h         | 3 +--
 support/support_capture_subprocess.c | 6 +++---
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/support/capture_subprocess.h b/support/capture_subprocess.h
index e44c965ef3df1521a164b63ffa50ddd7cf904ccc..037b9c220ebc34d2585716bcfcf1fa3d45255d94 100644
--- a/support/capture_subprocess.h
+++ b/support/capture_subprocess.h
@@ -44,8 +44,7 @@ struct support_capture_subprocess support_capture_subprogram
 /* Copy the running program into a setgid binary and run it with CHILD_ID
    argument.  If execution is successful, return the exit status of the child
    program, otherwise return a non-zero failure exit code.  */
-int support_capture_subprogram_self_sgid
-  (char *child_id);
+int support_capture_subprogram_self_sgid (const char *child_id);
 
 /* Deallocate the subprocess data captured by
    support_capture_subprocess.  */
diff --git a/support/support_capture_subprocess.c b/support/support_capture_subprocess.c
index a8bcb23d408d00c4804ce9569cffb965a916f7f0..1b4aa66ede5e98a14ef5c69f4d2c85bb5f33d7b6 100644
--- a/support/support_capture_subprocess.c
+++ b/support/support_capture_subprocess.c
@@ -109,7 +109,7 @@ support_capture_subprogram (const char *file, char *const argv[])
    safely make it SGID with the TARGET group ID.  Then runs the
    executable.  */
 static int
-copy_and_spawn_sgid (char *child_id, gid_t gid)
+copy_and_spawn_sgid (const char *child_id, gid_t gid)
 {
   char *dirname = xasprintf ("%s/tst-tunables-setuid.%jd",
 			     test_dir, (intmax_t) getpid ());
@@ -172,7 +172,7 @@ copy_and_spawn_sgid (char *child_id, gid_t gid)
   ret = 0;
   infd = outfd = -1;
 
-  char * const args[] = {execname, child_id, NULL};
+  char * const args[] = {execname, (char *) child_id, NULL};
 
   status = support_subprogram_wait (args[0], args);
 
@@ -199,7 +199,7 @@ err:
 }
 
 int
-support_capture_subprogram_self_sgid (char *child_id)
+support_capture_subprogram_self_sgid (const char *child_id)
 {
   gid_t target = 0;
   const int count = 64;
-- 
2.53.0

